docs/guides/snowflake_integration.md

```diff
@@ -486,3 +513,126 @@ FN_T_UID2_IDENTITY_MAP(EMAIL_HASH, 'email_hash')
+
+A sharing participant must encrypt [raw UID2s](../ref-info/glossary-uid#gl-raw-uid2) into [UID2 tokens](../ref-info/glossary-uid#gl-uid2-token) before sending them to another participant.
+
+The following functions support UID2 sharing:
+
+- [`FN_UID2_ENCRYPT`](#fn_uid2_encrypt)
+- [`FN_UID2_DECRYPT`](#fn_uid2_decrypt)
+
+### FN_UID2_ENCRYPT
+
+This function encrypts a raw UID2 and returns the corresponding UID2 token.
+
+Use the applicable prefix to indicate your role:
+- For advertisers: `ADV.FN_UID2_ENCRYPT`
+- For data providers: `DP.FN_UID2_ENCRYPT`
+
+>NOTE: The raw UID2 value is a Base64-encoded string, 44 characters in length.
+
+#### FN_UID2_ENCRYPT: Syntax
+
+The following is the basic syntax for the encrypt function:
+
+```
+Function: FN_UID2_ENCRYPT(raw_uid VARCHAR(44))
+RETURNS VARCHAR(220)
+```
+
+#### FN_UID2_ENCRYPT: Example
+
+The following example illustrates use of this function.
+
+```
+SELECT T.raw_uid, LATERAL (FN_UID2_ENCRYPT(T.raw_uid)) AS token
+FROM AUDIENCE_WITH_UID2 T
+```
+
+Result: An additional column named `token` is added to the query result.
+
+```
++------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| raw_uid                                                          | token                                                                                                                                                                                                                        |
+|------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| 0123456789ABCDEF0123456789ABCDEF0123456789ABCDEF0123456789ABCDEF | AAMAAAACAAECAwQFBgcICQoL18RtKOdx1LlG2zgxK7bTW/nomJIeMFyEJOAMaer+20FTzLBNCJRajrXzXWzb9H3FCtTXs5Rl8PJhYnxxWmMtQ11VHfjI7XiA71CctX2vvCBzD6VEszdNq/t66B1cvLKVjoq2lijm6yyaRi0ovJPuk2dcb8dm4dkHMWGkkOrXbPPvrpsIvYQHH11sbqVycYJZm5w= |
++------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+```
+
+### FN_UID2_DECRYPT
+
+This function decrypts a UID2 token and returns the corresponding raw UID2.
+
+Use the applicable prefix to indicate your role:
+- For advertisers: `ADV.FN_UID2_DECRYPT`
+- For data providers: `DP.FN_UID2_DECRYPT`
+
+#### FN_UID2_DECRYPT: Syntax
+
+The following is the basic syntax for the decrypt function:
+
+```
+Function: FN_UID2_DECRYPT(token VARCHAR(220))
```

Comment:
Similar to the comment above about FN_UID2_ENCRYPT:

#### FN_UID2_DECRYPT: Syntax

```
FN_UID2_DECRYPT(token VARCHAR)
```

#### Arguments

|Argument|Data Type|Description|
| :--- | :--- | :--- |
| `token` | varchar | The advertising token. |

A successful query returns the raw UID2 as a `VARCHAR`.

Reply:
Done, thx!